src/components/data/test/Thumbnail-test.js

```diff
@@ -0,0 +1,53 @@
+import { mount } from 'enzyme';
+
+import { testCompositeComponent } from '../../test/common-tests';
+
+import Thumbnail from '../Thumbnail';
+
+const createComponent = (Component, props = {}) => {
+  return mount(<Component {...props} />);
+};
+
+describe('Thumbnail', () => {
+  testCompositeComponent(Thumbnail);
+
+  it('disables container border when `borderless` is set', () => {
+    const withBorder = createComponent(Thumbnail);
+    const withoutBorder = createComponent(Thumbnail, { borderless: true });
+
+    // The "border" on Thumbnails is formed by padding classes. Default
+    // padding size ("md" size) is `p-3`. The absence of padding means there
+    // is no border.
+    assert.isTrue(
+      withBorder.find('[data-composite-component="Thumbnail"]').hasClass('p-3')
+    );
+    assert.isFalse(
+      withoutBorder
+        .find('[data-composite-component="Thumbnail"]')
+        .hasClass('p-3')
```

Comment:
This test would ideally check for element not having any padding, rather than not having a specific padding. If `borderless` caused the padding to be set to some other value for example, this would pass.